File: src/client/lourd/src/main/java/com/cyberacy/negotrack/models/Session.java
package com.cyberacy.negotrack.models;

import com.cyberacy.negotrack.models.entities.Account;

public class Session {

    private static Session instance = null;
    final private Account account;

    private Session(Account account) {
        this.account = account;
    }

    public static void openSession(Account account) {
        if(instance == null) {
            instance = new Session(account);
        }
    }

    public static void closeSession() {
        instance = null;
    }

    public static Session getInstance() {
        return instance;
    }

    public Account getAccount() {
        return account;
    }

}

File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/Session.kt
package com.cyberacy.app.models.entities


class Session private constructor(private val person: Person) {

    companion object {
        private var instance: Session? = null

        fun openSession(person: Person) {
            if(instance == null) {
                instance = Session(person)
            }
        }

        fun closeSession() {
            instance = null
        }

        fun getSession(): Session? {
            return instance
        }

        fun getJwtToken(): String {
            if(instance != null) {
                return instance!!.person.token ?: ""
            }
            return ""
        }
    }





}